### ISSUE TITLE
activesupport requires Ruby version >= 2.5.0.

### ISSUE THREAD

**Issue by gileshinchcliff** [fabric8io/fluent-plugin-kubernetes_metadata_filter#184]
when installing this plugin like so:
```bash-4.2# td-agent-gem install fluent-plugin-kubernetes_metadata_filter              
ERROR:  Error installing fluent-plugin-kubernetes_metadata_filter:
	activesupport requires Ruby version >= 2.5.0.
```
Having done some debugging it's not straightforward to see where in the dependency chain activesupport is being used, however I can see it got updated 3 days ago.

My guess is it's not fixed to a version somewhere in the dependency chain, however I'd need some support to go further with this debug.

Thanks

Giles

**Comment by richm** [comment]
I have no idea.  The dependencies for this plugin haven't changed in a while:
```
  gem.add_runtime_dependency 'fluentd', ['>= 0.14.0', '< 2']
  gem.add_runtime_dependency "lru_redux"
  gem.add_runtime_dependency "kubeclient", "~> 1.1.4"
```
If you figure it out, please let us know.

**Comment by gileshinchcliff** [comment]
Can you recreate the issue?

**Comment by richm** [comment]
I have not tried

**Comment by abhishek-buragadda** [comment]
I am also facing this issue.  below is my docker file



> FROM fluent/fluentd:v0.12
> 
> USER root
> 
> RUN apk add --update --virtual .build-deps \
>         sudo build-base ruby-dev \
>  && sudo -u root  gem install fluent-plugin-kubernetes_metadata_filter
> 
> EXPOSE 24224

**Comment by richm** [comment]
See https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/pull/185

**Comment by paigr** [comment]
I did some quick investigation into this; it turns out the version of `kubeclient` this plugin depends has an unpinned dependency on `activesupport`, you can see this here https://rubygems.org/gems/kubeclient/versions/1.1.4. `activesupport` has recently released version 6.0.0 which adds the `zeitwerk` dependency.

**Comment by richm** [comment]
ok.  The latest version of this plugin is https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/releases/tag/v2.3.0 which gets rid of the dependency on activesupport.

**Comment by Ewocker** [comment]
Seeing zeitwork issue as well with fluentd v0.12 same as @stoyta 
No changes since last succesfully build either.

**Comment by richm** [comment]
> Seeing zeitwork issue as well with fluentd v0.12 same as @stoyta
> No changes since last succesfully build either.

Can you use fluentd 1.x instead of 0.12?  Then you can use v2.3.0 of this plugin which should eliminate the problem

**Comment by amaraaon** [comment]
> We are seeing a similar issue but different library using fluentd:v0.12.
> 
> ERROR: Error installing fluent-plugin-kubernetes_metadata_filter:
> zeitwerk requires Ruby version >= 2.4.4.
> 
> No changes have been made to the dockerfile since the last successful build.

rbenv install -v 2.4.4
rbenv global 2.4.4

**Comment by jcantrill** [comment]
fixed by #185
